nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -562,13 +547,6 @@ Costmap2DROS::start()
     stopped_ = false;
   }
   stop_updates_ = false;
-
-  // block until the costmap is re-initialized.. meaning one update cycle has run
-  rclcpp::Rate r(20.0);
-  while (rclcpp::ok() && !initialized_) {
```

Comment:
I think we have the `initialized_` member to ensure we trigger at least 1 time before we do other things which has been removed. This ensures that we have some data to look at and not blank-defaults if a request comes in sooner after startup and/or there is a very slow update rate. So to remove that, we need to make sure updateMap is triggered at some point during the lifecycle activation state (or readd).

Reply:
As I see from the code, currently `initialized_` indicator is being used to prevent uninitialized map usage from `Costmap2DROS::start()` and `Costmap2DROS::resume()`.  The `start()` routine is being called from `Costmap2DROS::on_activate()` callback (or `start()` also could be called manually, as in public domain).
In both cases in `start()` and in `resume()`, we are waiting in for a `mapUpdateLoop()` loop will call `updateMap()` and thus  set `initialized_` to `true`. Instead of this we could remove these waits and just manually call `mapUpdateLoop()` routine, which will set all maps.

There are remain setting of `initialized_ = false` in `stop()` and `pause()` routines. If `pause()` was called, `initialized_` variable is used only in `start()` and in `resume()`, where we will intentionally update maps instead, so this is also not an issue.
As for the `stop()` is being called, `initalized_` is set to `false` and we need re-initialization. But since is `stopped_ = false` too, `mapUpdateLoop()` will be skipped and maps are not updated. This is not an issue, if call `start()`, but won't work for `resume()` routine.

In other words, possible pitfall we could get is: call `resume()` after `stop()` if apply this patch. @SteveMacenski, did you mean this case, or I've missed something else?